src/main/java/io/aiven/kafka/connect/common/output/OutputWriter.java

```diff
@@ -118,21 +132,21 @@ public Builder withOutputFields(final Collection<OutputField> outputFields) {
             return this;
         }
 
-        public Builder withOutputStream(final OutputStream outputStream) {
-            this.outputStream = outputStream;
-            return this;
-        }
-
-        public OutputWriter build(final FormatType formatType) throws IOException {
+        public OutputWriter build(final OutputStream out, final FormatType formatType) throws IOException {
             Objects.requireNonNull(outputFields, "Output fields haven't been set");
-            Objects.requireNonNull(outputStream, "Output stream hasn't been set");
+            Objects.requireNonNull(out, "Output stream hasn't been set");
             switch (formatType) {
                 case CSV:
-                    return new PlainOutputWriter(outputFields, getCompressedStream(outputStream));
+                    return new PlainOutputWriter(outputFields, getCompressedStream(out));
                 case JSONL:
-                    return new JsonLinesOutputWriter(outputFields, getCompressedStream(outputStream));
+                    return new JsonLinesOutputWriter(outputFields, getCompressedStream(out));
                 case JSON:
-                    return new JsonOutputWriter(outputFields, getCompressedStream(outputStream));
+                    return new JsonOutputWriter(outputFields, getCompressedStream(out));
+                case PARQUET:
+                    if (Objects.isNull(externalProperties)) {
+                        externalProperties = Collections.emptyMap();
+                    }
+                    return new ParquetOutputWriter(outputFields, out, externalProperties);
```

Comment:
We rely on Parquet's own compression, right? Maybe it's worth commenting here.

Reply:
Also, on the connector level it's worth adding a warning to the logs if the compression is and Parquet is used.